### PR TITLE
[#184] Typescript checks in CI 

### LINKED ISSUE #184: Improve Typescript checks and utiities
# Clarification and motivation

In #154 the typescript API has been introduced, however there's two things that could be improved.

First: while CI checks that the API builds correctly, it does not check that the API has been re-generated to match code changes (when there are any).
We should verify that, I suggest doing so as in the `check cabal files` step, so by running the re-generation and check for no diff.
Additionally: the TS build step runs on scheduled builds as well, which is likely unnecessary, so we should reconsider that.

Second point: there's two small improvements that we could use in the top-level `Makefile`:

1. we should provide the env vars to `stack build` in the `typescript` target (see e.g. the `test` one)
Otherwise there may be issues when building the Haskell library for the first time.

2. consider removing the typescript dir "clearing" from the `clean` target.
That command was only used for git-ignored results of Ligo compilation, I think it should go back to that as the typescript stuff gets committed. Moreover, there is already a "cleaning" step in the `typescript` target, so this seems unnecessary.

# Acceptance criteria

1. CI checks that the TS API matches the one generated from the code.
2. (if deemed useful) the existing API build step is prevented from running on scheduled builds
3. the default values for the contract and storage(s) env-var are given to the `./Makefile` `typescript` target
4. (if deemed useful) the `clean` target does not remove the generated typescript files

### PR DESCRIPTION
**Pull request by sras**
## Description

Adds checks to ensure that the typescript files have been regenerated to match any change to contract interface.

## Related issue(s)

Resolves #184 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -102,7 +102,16 @@ steps:
   #     ./result/bin/baseDAO-test --nettest-run-network
   #       --pattern '\$1 == "On network" || \$NF == "On network" || \$0 ~ /.On network./'
 
+  - label: check typescript api
+    depends_on:
```
Comment (pasqu4le) [inline .buildkite/pipeline.yml:106]: Perhaps this too should run on master only after a PR is merged and not in a scheduled run?
```suggestion
    if: *not_scheduled
    depends_on:
```

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -102,7 +102,16 @@ steps:
...
+      - build-ligo
+      - build-haskell
+    commands:
+      - nix-build ci.nix -A packages.baseDAO-ligo-meta.tests.baseDAO-test
```
Comment (pasqu4le) [inline .buildkite/pipeline.yml:110]: AFAICT this step is unnecessary, as the `baseDAO-ligo-meta` package is already built by `build-haskell` and tests aren't necessary to check the typescript diff.

**Review by pasqu4le — COMMENTED** [review]
Two small comments, but it looks like the API is already outdated, so we need to fix that here as well.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (sras) [reply .buildkite/pipeline.yml]: Turned out to be true. Have removed it.

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**